test/shed_functional/base/twilltestcase.py

```diff
@@ -261,12 +261,16 @@ def commit_and_push( self, repository, hgrepo, options, username, password ):
         commands.commit( ui.ui(), hgrepo, **options )
         try:
             commands.push( ui.ui(), hgrepo, dest=url )
-        except Abort as a:
-            message = a
-            if 'authorization failed' in message:
+        except Exception as e:
+            if str(e).find('Pushing to Tool Shed is disabled') != -1:
                 return False
             else:
-                raise
+                return self.__push_again( hgrepo, url )
+        return True
+
+    def __push_again( hgrepo, url ):
+        try:
+            commands.push( ui.ui(), hgrepo, dest=url )
         except Exception as e:
             if str(e).find('Pushing to Tool Shed is disabled') != -1:
                 return False
```

Comment:
This is returning `Null` if `commands.push()` succeeds, which is probably not what you want.

Maybe use a `for` cycle in `commit_and_push()`:
```
        for i in range(2):
            try:
                commands.push( ui.ui(), hgrepo, dest=url )
            except Exception as e:
                if str(e).find('Pushing to Tool Shed is disabled') != -1:
                    return False
            else:
                return True
        raise
```